### PR TITLE
Update manifest

### PR DESCRIPTION
**Pull request by XX-Yin**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
1. Update the detection of stream modality.
2. Save the manifest file every trial. 

### What issues or discussions does this update address?
1. resolved: https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/1114#issuecomment-2515779489

2. The manifest file should be updated every trial as new data stream may be added after start of the session. 

### Describe the expected change in behavior from the perspective of the experimenter
No

### Describe any manual update steps for task computers
No

### Was this update tested in 446/447?
tested on 323_Ephys3

### CODE REVIEW

File: src/foraging_gui/Foraging.py [hunk 1]
```diff
@@ -223,9 +222,6 @@ def __init__(self, parent=None,box_number=1,start_bonsai_ide=True):
         # Initializes session log handler as None
         self.session_log_handler = None
 
-        # generate an upload manifest when a session has been produced
-        self.upload_manifest_slot = self.sessionGenerated.connect(self._generate_upload_manifest)
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:227]: Why remove the signal/slot architecture Micah added?

Comment (XX-Yin) [reply src/foraging_gui/Foraging.py:227]: I guess that structure is used for running the "_generate_upload_manifest" only once, which could be problematic (e.g. when we start the video recording after the first trial, the video modality will not be recorded.)

File: src/foraging_gui/Foraging.py [hunk 2]
```diff
@@ -4628,7 +4625,7 @@ def _open_mouse_on_streamlit(self):
                          '&session_plot_selected_draw_types=1.+Choice+history'
         )
 
-    def _generate_upload_manifest(self, session: Session):
+    def _generate_upload_manifest(self, generated_medadata):
```
[the remark this answers is not in the document]
Comment (XX-Yin) [reply src/foraging_gui/Foraging.py:4628]: It's just safer to check the modality and generate the manifest file each trial, which is not computation intensive.

**Comment by alexpiet** [comment]
Can you make a simplified PR that just adds the correct modality to the manifest? I don't think we should change the way the signal/slot architecture is used, or update the manifest on every trial. Especially without consulting @micahwoodard and the incorrect manifest is going to effect all the sessions on monday

**Comment by XX-Yin** [comment]
> Can you make a simplified PR that just adds the correct modality to the manifest? I don't think we should change the way the signal/slot architecture is used, or update the manifest on every trial. Especially without consulting @micahwoodard and the incorrect manifest is going to effect all the sessions on monday

We can wait for the feedback from @micahwoodard. This PR doesn't change the manifest file. It just updates the manifest file every trial.

**Comment by alexpiet** [comment]
Well, no. The `develop` branch has a bug that will cause all data to fail to upload: https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/1114

If we don't fix it this week then we can't merge the updates (3rd week in a row...)

**Comment by XX-Yin** [comment]
So, we should review this PR ASAP. If you still have concerns after I tested it, you can also test it on your own computer for double check.

**Comment by alexpiet** [comment]
I'm asking you to remove everything that isn't essential to fixing the bug

**Comment by XX-Yin** [comment]
> I'm asking you to remove everything that isn't essential to fixing the bug

Please feel free to make commits to this PR to remove the code you think that is not essential. They all look essential to me.

**Comment by XX-Yin** [comment]
Then let's merge this PR at the end of this week!

**Comment by alexpiet** [comment]
Ok, I just made a new PR with only the essential fixes(https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/1122). We can close this PR

**Comment by alexpiet** [comment]
Ok, you are right that if you run a session with the camera, and then start a new session without the GUI, then it incorrectly lists behavior-video as a modality in the manifest and in the session metadata. I'll make a separate issue, since I don't have time to fix that right now. Lets merge this fix in, since it will effect every session

**Pull request closed without merging**